src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -1045,12 +1045,45 @@ private void set_package_names_if_all_is_specified(ChocolateyConfiguration confi
                 config.Input = string.Empty;
 
                 var localPackages = list_run(config, logResults: false);
+                var packagesToUpdate = localPackages.Select(p => p.Key).ToList();
+
+                if (!String.IsNullOrEmpty(config.UpgradeCommand.PackageNamesToSkip))
+                {
+                    var packagesToSkip = config.UpgradeCommand.PackageNamesToSkip
+                        .Split(',')
+                        .Select(x => x.Trim())
+                        .Where(x => !String.IsNullOrEmpty(x))
+                        .ToList();
+
+                    var unknownPackagesToSkip = packagesToSkip
+                        .Where(x => !packagesToUpdate.Contains(x, StringComparer.OrdinalIgnoreCase))
+                        .ToList();
+
+                    if (unknownPackagesToSkip.Any())
+                    {
+                        this.Log().Warn(() => "Some packages specified in '-except' were not found in the local packages: '{0}'".format_with(String.Join(",", unknownPackagesToSkip)));

```

Comment:
I'm not sure I get what you mean with your last comment? The 'unknown' packages in the code are packages that were specified in the list of exceptions, but since they were not in the list of installed packages, they cannot be skipped :)

An example would be: `choco upgrade all --except="skyp"` - it's not a critical problem, but it's good that the user knows that `Skype` will be upgraded because `skyp` was not in the installed packages (typo).

Do you have a suggestion for a better message or behavior?